### ISSUE TITLE
Problems when I activate Live Unit Testing using NUnit in a Visual Studio 2017 Test Project

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit3-vs-adapter#536]
@licjapodaca commented on [Sun Aug 05 2018](https://github.com/nunit/nunit/issues/2949)

I'm using Visual Studio 2017 Enterprise 15.7.6 and a Test Project with the following Nuget Packages:

![image](https://user-images.githubusercontent.com/828781/43696144-829ebcbc-98f1-11e8-91b6-eec4df2ce9ba.png)

The problem is that when I activate **Live Unit Testing (LUT)** in VS2017 and specify in a Test Method the attribute [TestCase] in some cases works the LUT but when I mke some change in source code other Test Methods are excluded from LUT as you can see in the next image:

![image](https://user-images.githubusercontent.com/828781/43696328-492125aa-98f2-11e8-897c-77b7fc149b23.png)

The source code that has the TestCase attribute is the following:

```csharp
[Test]
[TestCase(null), TestCase(""), TestCase(" ")]
public void Log_InvalidError_ThrowArgumentNullException(string error)
{
	var logger = new ErrorLogger();

	Assert.That(() => logger.Log(error), Throws.ArgumentNullException);
}
```

> NOTE: If I stop and start again LUT in VS2017 the Test Method mentioned at the beginning of this ISSUE is included correctly, but again if I refactor the source code, the LUT exclude the same Test Method again.

Is there a workaround for this problem?

Thanks for your help in advance.

---

@licjapodaca commented on [Tue Aug 07 2018](https://github.com/nunit/nunit/issues/2949#issuecomment-411270384)

Anybody?

---

@CharliePoole commented on [Tue Aug 07 2018](https://github.com/nunit/nunit/issues/2949#issuecomment-411276054)

Moving this to the nunit3 vs adapter repository, where it's likely to get more attention.

**Comment by licjapodaca** [comment]
Thanks @CharliePoole for your help...

**Comment by OsirisTerje** [comment]
@licjapodaca  Can you try this with 15.8 preview 5?    (Or create a small repro zip and I can check that)

**Comment by licjapodaca** [comment]
Hi @OsirisTerje nice to meet you, about the trying with 15.8 preview, the company were I work doesn't let me to install Beta or Previews releases on the company developers PCs... only versions that already released as a production... Do you know when is going to be available the 15.8 version as a production release?

**Comment by OsirisTerje** [comment]
It should be due soon.  The minor release rhythm is approximately 6 weeks (see https://docs.microsoft.com/en-us/visualstudio/productinfo/vs2017-release-rhythm)  and 15.7 was released May 7th, so it should not be long.  They never give fixed dates though, so the rhythm is all we ave to go by.  
But, if you create a small repro I can check the issue.
